### PR TITLE
Add NOTICE file

### PR DESCRIPTION
**Pull request by mundaym**
Add a NOTICE file containing some information about the project.

### CODE REVIEW

**Review by rswarbrick — APPROVED** [review]

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,10 @@
+The Ibex Project
+Copyright 2024 lowRISC contributors.
+
+This product includes hardware and/or software developed as part of the
+Ibex(R) (https://github.com/lowRISC/ibex) and OpenTitan(R) projects.
```
Comment (nbdd0121) [inline NOTICE:5]: Do we want (R) or ®?

File: NOTICE [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+Ibex was originally developed by the PULP team at ETH Zurich and University of
+Bologna under the name zero-riscy. Ibex verification, performance enhancement
+and security hardening have been supported by the OpenTitan project
```
Comment (luismarques) [inline NOTICE:9]: regarding "performance enhancement[s]", I guess we also added features? (and might add more in the future)

File: NOTICE (hunk 1 above)
Comment (mundaym) [reply NOTICE:5]: Unfortunately ® is rejected by our CI system at the moment as it is not ASCII.

File: NOTICE (hunk 2 above)
Comment (mundaym) [reply NOTICE:9]: Addressed in the OT repo review.

**Pull request merged**